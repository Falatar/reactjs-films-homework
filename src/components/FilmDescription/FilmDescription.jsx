import React from 'react';
import PropTypes from 'prop-types';
import style from './FilmDescription.scss';
import styleAlter from './FilmDescriptionAlter.scss';

function FilmDescription({
  name, tagList, rating, mode,
}) {
  const scss = mode ? style : styleAlter;
  return (
    <div className={scss.description}>
      <div className={scss.title}>
        <h3 className={scss.name}>{name}</h3>
        <h2 className={scss.tags}>{tagList}</h2>
      </div>
      <h1 className={scss.rate}>{rating.toFixed(1)}</h1>
    </div>
  );
}

FilmDescription.defaultProps = {
  name: 'Can\'t find property "name"',
  tagList: ['unknown'],
  rating: 0,
  mode: true,
};

FilmDescription.propTypes = {
  name: PropTypes.string,
  tagList: PropTypes.string,
  rating: PropTypes.number,
  mode: PropTypes.bool,
};

export default FilmDescription;
