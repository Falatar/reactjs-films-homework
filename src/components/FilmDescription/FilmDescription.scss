.description {
    width: 100%;
    margin: 0;
    padding: 0;
    border: 0;
    height: 20%;
    display: flex;
    flex-direction: row;
    align-items: center;

    .title {
        width: 75%;
        margin-left: 5%;
        padding: 0;
        border: 0;
        display: flex;
        flex-direction: column;
        justify-content: center;

        .name {
            width: 100%;
            text-transform: uppercase;
            text-overflow: ellipsis;
            overflow: hidden;
            white-space: nowrap;
            font-weight: 600;
            font-size: 1.6em;
            color: black;
            margin: 0;
            padding: 0;
            border: 0;
        }
        
        .tags {
            text-transform: uppercase;
            text-overflow: ellipsis;
            overflow: hidden;
            white-space: nowrap;
            font-size: 1em;
            width: 100%;
            color: deepskyblue;
            margin: 0;
            padding: 0;
            border: 0;
        }
    }

    .rate {
        height: 1.1em;
        width: 2em;
        font-size: 1.4em;
        margin: 0 0 0 0.5em;
        padding: 0;
        border: 1px solid;
        color: deepskyblue;
        border-color: deepskyblue;
        text-align: center;
    }
}