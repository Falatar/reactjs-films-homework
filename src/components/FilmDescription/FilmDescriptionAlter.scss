.description {
    display: flex;
    flex-direction: row;
    width: 100%;
    height: 20%;

    .title {
        display: flex;
        flex-direction: column;
        width: 70%;
        height: 100%;
        margin-right: 5%;
        margin-left: 3%;

        .name {
            color: white;
            z-index: 2;
            margin: 0;
            margin-bottom: 1%;
        }
        
        .tags {
            z-index: 2;
            text-transform: uppercase;
            text-overflow: ellipsis;
            overflow: hidden;
            white-space: nowrap;
            font-size: 1em;
            width: 100%;
            color: deepskyblue;
            margin: 0;
            padding: 0;
            border: 0;
        }
    }

    .rate {
        height: 1.1em;
        width: 2em;
        font-size: 1.4em;
        margin: 0 0 0 0.5em;
        padding: 0;
        border: 1px solid;
        color: deepskyblue;
        border-color: deepskyblue;
        text-align: center;
        z-index: 2;
    }
}