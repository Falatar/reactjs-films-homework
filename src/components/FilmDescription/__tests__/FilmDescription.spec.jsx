import React from 'react';
import renderer from 'react-test-renderer';
import FilmDescription from '../FilmDescription';

describe('FilmDescription', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<FilmDescription name="nemo" genrePtime="empty" rating={0.0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly in alter mode', () => {
    const tree = renderer
      .create(<FilmDescription name="nemo" genrePtime="empty" rating={0.0} mode={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
