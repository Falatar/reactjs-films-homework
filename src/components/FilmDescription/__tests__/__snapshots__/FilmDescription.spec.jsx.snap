// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDescription renders correctly 1`] = `
<div
  className="description"
>
  <div
    className="title"
  >
    <h3
      className="name"
    >
      nemo
    </h3>
    <h2
      className="tags"
    >
      unknown
    </h2>
  </div>
  <h1
    className="rate"
  >
    0
  </h1>
</div>
`;

exports[`FilmDescription renders correctly in alter mode 1`] = `
<div
  className="description"
>
  <div
    className="title"
  >
    <h3
      className="name"
    >
      nemo
    </h3>
    <h2
      className="tags"
    >
      unknown
    </h2>
  </div>
  <h1
    className="rate"
  >
    0
  </h1>
</div>
`;
